src/heap-builder.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import {AllocationProfileNode} from './v8-types';
+import {perftools} from './profile';
+
+// A stack of function UIDs.
+// TODO: duplicated in builder.ts
+type Stack = Array<number>;
+
+
+// TODO: pass this value explicitly when starting the profiler
+const SAMPLE_PERIOD = 512 * 1024;
+
+let strings = [''];
+let samples: Array<perftools.profiles.Sample> = [];
+
+let locationIds = ['dummy'];
+let locations: Array<perftools.profiles.Location> = [];
+let locationMap: Map<number, perftools.profiles.Location>;
+
+let functionIds = ['dummy'];
+let functions: Array<perftools.profiles.Function> = [];
+let functionMap: Map<number, perftools.profiles.Function>;
+
+function getIndexOrAdd<T>(element: T, lst: Array<T>) {
```

Comment:
Should probably add a TODO to merge this with the same function in the other file as part of a `util` file.

Reply:
Done.